File: cpp/contains_duplicate.cc
/*
 * Filename: contains_duplicate.cc
 * Author: Jun Xu
 * Email: jxu.vimacs@gmail.com
 * Created Time: Fri 14 Aug 2015 12:40:34 PM EDT
 */

#include <iostream>
using namespace std;

class Solution {
	public:
		bool containsDuplicate(vector<int>& nums) {
			if (nums.size() <= 1)
			{
				return false;
			}

			unordered_set<int> set;

			for (auto i = 0; i < nums.size(); i++)
			{
				if (set.find(nums[i]) != set.end())
				{
					return true;
				}
				else
				{
					set.insert(nums[i]);
				}
			}

			return false;
		}
};

File: java/ContainsDuplicate.java
/*
 * Filename: ContainsDuplicate.java
 * Author: Jun Xu
 * Email: jxu.vimacs@gmail.com
 * Created Time: Thu 04 Jun 2015 12:27:50 PM CST
 */

public class Solution {
	public boolean containsDuplicate(int[] nums) {
		if (nums == null || nums.length <= 1) {
			return false;
		}

		HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
		for (int i = 0; i < nums.length; i++) {
			if (map.contains(nums[i])) {
				return true;
			} else {
				map.put(nums[i], i);
			}
		}

		return false;
	}
}
